core/ustring.cpp

```diff
@@ -1814,14 +1810,14 @@ int64_t String::bin_to_int64(bool p_with_prefix) const {
 		if (c == '0' || c == '1') {
 			n = c - '0';
 		} else {
-			return 0;
+			ERR_FAIL_COND_V_MSG(true, 0, "Invalid binary notation character \"" + chr(*s) + "\" in string \"" + *this + "\".");
```

Comment:
Could be a simple `ERR_PRINT` and `return 0;`, otherwise it will show the reason for failure as "Condition 'true' is true."

Alternatively, reverse the order:
```cpp
ERR_FAIL_COND_V_MSG(c != '0' && c != '1', 0, "Invalid binary notation character \"" + chr(*s) + "\" in string \"" + *this + "\".");
n = c - '0';
```

Reply:
Or just `ERR_FAIL_V_MSG(...)` without cond.